app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt

```diff
@@ -216,9 +224,51 @@ class ChatRoomFragment : Fragment(), ChatRoomView, EmojiKeyboardListener, EmojiR
     override fun onActivityResult(requestCode: Int, resultCode: Int, resultData: Intent?) {
         if (requestCode == REQUEST_CODE_FOR_PERFORM_SAF && resultCode == Activity.RESULT_OK) {
             if (resultData != null) {
-                uploadFile(resultData.data)
+                fileAttachmentDialog(resultData.data)
+            }
+        }
+    }
+
+    private fun fileAttachmentDialog(data: Uri) {
+        val builder = AlertDialog.Builder(activity)
+        val dialogView = View.inflate(context, R.layout.file_attachments_dialog, null)
+        builder.setView(dialogView)
+        val alertDialog = builder.create()
+
+        dialogView?.let {
+            val imagePreview: ImageView = it.findViewById(R.id.image_preview)
```

Comment:
Don't need to change, but a more Kotlin idiomatic syntax would be:
`val imagePreview = it.findViewById<ImageView>(R.id.image_preview)`